sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1473,11 +1509,20 @@ pi_result piextDeviceCreateWithNativeHandle(pi_native_handle NativeHandle,
   assert(Device);
   assert(Platform);
 
-  // Create PI device from the given Level Zero device handle.
-  // TODO: get the device from the devices' cache.
+  std::lock_guard<std::mutex> Lock(Platform->PiDevicesCacheMutex);
+  populateDeviceCacheIfNeeded(Platform);
+
+  // Scan the cache for the requested device handle. If it is not present in the
+  // cache an error will be returned for an invalid value for the NativeHandle.
```

Comment:
Same comment here about describing "why" in the comment.

Reply:
Done.